elastic/logstash
# Load a plugin by alias name.

If the plugin name is an alias and no other plugin exists with same name then load the alised one



## Release notes

Introduce the concept of aliased plugin, to let a plugin be renamed easily

## What does this PR do?



An alias for a plugin is a just a rename of an existing plugin, in this way the alias can be used in pipeline configuration. If later a real plugin with same name of the alias is released then that plugin takes precedence over the alias name. 
Suppose we a plugin filter named `teleport` and we also alias it as `teleport_over_fiber`, the `teleport_over_fiber` alias can be used in interchangeable way as `teleport` with same parameters and behavior. In a future when a real plugin that leverage fiber transmission for teleport a plugin `teleport_over_fiber` name is effectively released, this real plugin takes precedence to over the alias.

The changes to `logstash-plugin` tool used to manage the installation, listing and removal of aliased plugin is handled by the PR #12821

## Why is it important/What is the impact to the user?


It lets the user to star using today plugins names that will be released as independent plugin in near future.

## Checklist




## Author's Checklist


- [x] try a local run

## How to test this PR locally


Use an alias in a pipeline config, for example:
```ruby
input {
  elastic_agent { # beats alias
     port => 5044
   }
}
output { stdout {} }
```
check that it runs correctly with `bin/logstash -f pipeline.conf`

## Related issues


- 

## Use cases



## Screenshots



## Logs
